### ISSUE TITLE
Allow for offset and size information display simultaneously 

### ISSUE THREAD

**Issue by BenBE** [lsof-org/lsof#165]
`lsof -P -F` should list both `o` (offset) and `s` (size) for regular files. Currently this only lists the file size (`s`), but when `lsof -P -o -F` also requests the offset information the size information is dropped. There's no (sane) way to get `lsof` to report both information at the same time …

Given the way data output [works](https://github.com/lsof-org/lsof/blob/234ac9ca6082084640cb1b762f6721eccc993c79/print.c#L1003-L1074) this looks like someone seriously wants this usecase to be broken …

**Comment by masatake** [comment]
I agree. However, as far as, reading the man page of lsof, this brokenness (or limitation) may be by design.

```
SIZE, SIZE/OFF, or OFFSET
...
                 The -o and -s options are mutually exclusive; they can't both be specified.
```

I wonder from where this limitation comes. As a user of GNU/Linux, I cannot understand why this limitation is needed.
I will accept the pull request removing this limitation as far as the man page and related documentation are updated, and the change is considered well about compatibility. However, I myself will not work on this topic.

If your main platform is GNU/Linux, I would like you to try lsfd (https://github.com/karelzak/util-linux/pull/1418) that I'm working on these days:

```
# ./lsfd -o +SIZE,POS -Q '(COMMAND == "auditd") and (ASSOC =~ "^[0-9]+$") and (TYPE == "REG") and (POS > 0) and (SIZE > 0)' 
COMMAND   PID USER ASSOC MODE TYPE SOURCE MNTID   INODE NAME                        SIZE     POS
auditd    864 root     6  -w-  REG   dm-0   118 1179745 /var/log/audit/audit.log 5770607 5770607
# ./lsfd -J -o +SIZE,POS -Q '(COMMAND == "auditd") and (ASSOC =~ "^[0-9]+$") and (TYPE == "REG") and (POS > 0) and (SIZE > 0)' 
{
   "lsfd": [
      {
         "command": "auditd",
         "pid": 864,
         "user": "root",
         "assoc": "6",
         "mode": "-w-",
         "type": "REG",
         "source": "dm-0",
         "mntid": 118,
         "inode": 1179745,
         "name": "/var/log/audit/audit.log",
         "size": 5772734,
         "pos": 5772734
      }
   ]
}
```

NOTE: the functions for getting information about the socket are still poor. After writing enough test cases for lsfd, I will work on the aspect.

**Comment by masatake** [comment]
The PR is welcome. Thank you.

>  In particular since htop requires the machine readable backend, which AFAIK lsfd does not provide.

Please, look at the example in my comment. lsfd has JSON output mode.

> OT: Is there some library that one could link against to use lsfd?

No. However, it provides JSON output mode. So if you have a JSON parser/reader, I guess using it from htop may not be hard.
